cli/client.go

```diff
@@ -1732,13 +1757,188 @@ func (conf *Config) checkUpdate(c *cli.Context) error {
 	// the local build is more than a week old, so we should warn
 	if time.Since(dateCompiled) > time.Hour*24*7 {
 		var updateInstructions string
-		if latestVersion != nil {
-			updateInstructions = fmt.Sprintf(" to version: %s", latestVersion.Original())
+		if latestBuildVersion.Original() != "" {
+			updateInstructions = fmt.Sprintf(" to version: %s", latestBuildVersion)
 		}
 		warningf(c.App.ErrWriter, "CLI Update Check: Your CLI is more than a week old. "+
-			"New CLI releases happen weekly; consider updating%s. See https://docs.viam.com/cli/#install", updateInstructions)
+			"New CLI releases happen weekly; consider updating%s. Run 'viam update' or see https://docs.viam.com/cli/#install", updateInstructions)
+	}
+	return nil
+}
+
+// UpdateCLIAction updates the CLI to the latest version.
+func UpdateCLIAction(c *cli.Context, args emptyArgs) error {
+	// 1. check CLI to see if update needed, if this fails then try update anyways
+	if latestBuildVersion, err := latestVersion(); err != nil {
+		if localVersion, err := localVersion(); err != nil {
+			if localVersion.LessThan(&latestBuildVersion) {
+				infof(c.App.Writer, "Your CLI is already up to date (version %s).", localVersion.Original())
+				return nil
+			}
+		}
+	}
+
+	// 2. check if cli managed by brew, if so attempt update. If it fails
+	// dont continue with binary to avoid putting brew out of sync
+	success, err := checkAndTryBrewUpdate()
+	if err != nil {
+		return errors.Errorf("CLI update failed: %v", err)
+	}
+	// brew update not returning false (not successful) but no error means not managed by brew
+	if !success {
+		// 3. get the local version binary path (use full path if no symlinks)
+		execPath, err := os.Executable()
+		if err != nil {
+			return errors.Errorf("CLI update failed: failed to get executable path: %v", err)
+		}
+		localBinaryPath, err := filepath.EvalSymlinks(execPath)
+		if err != nil {
+			localBinaryPath = execPath
+		}
+		directoryPath := filepath.Dir(localBinaryPath)
+
+		// 4. get the latest binary (from storage.googleapis.com) and write it into a temp file
+		binaryURL := binaryURL()
+		latestBinaryPath, err := downloadBinaryIntoDir(binaryURL, directoryPath)
+		if err != nil {
+			return errors.Errorf("CLI update failed: failed to download binary: %v", err)
+		}
+
+		// if replaceBinary succeeds, file is moved so Remove fails (harmless);
+		// if replaceBinary fails, file still exists so Remove cleans it up
+		defer os.Remove(latestBinaryPath) //nolint:errcheck
```

Comment:
Is it possible that we download a binary successfully but `downloadBinaryIntoDir` still returns an error for some other reason? It looks to me like that could happen. If so, we would never get to the `os.Remove` call.